api/v1beta2/kustomization_types.go

```diff
@@ -130,6 +134,8 @@ type KustomizationSpec struct {
 
 	// Timeout for validation, apply and health checking operations.
 	// Defaults to 'Interval' duration.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
```

Comment:
Would omit `h` for reasons specified in https://github.com/fluxcd/source-controller/pull/903#discussion_r975027656